src/etc/inc/gwlb.inc

```diff
@@ -2026,17 +2026,19 @@ function save_gateway($gateway_settings, $realid = "") {
 		isset($a_gateway_item[$realid]["nonlocalgateway"])) {
 		$realif = get_real_interface($a_gateway_item[$realid]['interface']);
 		$inet = (!is_ipaddrv4($a_gateway_item[$realid]['gateway']) ? "-inet6" : "-inet");
-		$cmd = "/sbin/route delete $inet " . escapeshellarg($a_gateway_item[$realid]['gateway']) . " -iface " . escapeshellarg($realif);
+		$rgateway = $a_gateway_item[$realid]['gateway'];
+		$cmd = "/sbin/route delete $inet " . escapeshellarg($a_gateway_item[$realid]['gateway']) . " -iface " . escapeshellarg($realif . " " . escapeshellarg($rgateway));
```

Comment:
`escapeshellarg($realif . " " . escapeshellarg($rgateway))` -- Is that intentional? `$rgateway` will be escaped and then the whole concatenated string escaped again. Looks to me like the closing paren for the first `escapeshellarg` needs moved to just after `$realif`, rather than doing it around both.

Reply:
typo, fixed